dell/xps/17-9700/common.nix

```diff
@@ -0,0 +1,29 @@
+{ lib, pkgs, ... }:
+
+{
+  # Boot loader
+  boot.loader.systemd-boot.enable = lib.mkDefault true;
+  boot.loader.efi.canTouchEfiVariables = lib.mkDefault true;
+  boot.kernelParams = lib.mkDefault [ "acpi_rev_override" ];
+  boot.kernelModules = lib.mkDefault [ "kvm-intel" ];
+
+  hardware.enableRedistributableFirmware = lib.mkDefault true;
+
+  # This will save you money and possibly your life!
+  services.thermald.enable = lib.mkDefault true;
+
+  boot.kernelPatches = [{
+    name = "enable-soundwire-drivers";
+    patch = null;
+    extraConfig = ''
+      SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES y
+      SND_SOC_INTEL_SOUNDWIRE_SOF_MACH m
+      SND_SOC_RT1308 m
```

Comment:
We can also enable those by default in nixpkgs....

Reply:
I would assume that they are not already there for a reason, but not really sure.

Reply:
I don't think so. Many kernel options are simply not enabled by default. People enable them as they need them.
Since it's just an optional module, this should be fine.

Reply:
> We can also enable those by default in nixpkgs....

I'm a little lost as to where these are set in nixpkgs. Seems like the builder lives here: https://github.com/ivanmalison/nixpkgs/blob/c159b9d51e4bd7a3d29744ab1c6baa78c71bf93b/pkgs/os-specific/linux/kernel/generic.nix#L79

but its not clear that anything gets passed by default.

Reply:
They can be enabled here: https://github.com/IvanMalison/nixpkgs/blob/c159b9d51e4bd7a3d29744ab1c6baa78c71bf93b/pkgs/os-specific/linux/kernel/common-config.nix

Reply:
This change in nixpkgs is not mandatory to merge this PR so.